AdobeStockImageAdminUi/view/adminhtml/web/js/components/grid/column/preview/actions.js

```diff
@@ -204,19 +204,20 @@ define([
                     success: function (response) {
                         var confirmationContent = $.mage.__('License "' + record.title + '"'),
                             quotaMessage = response.result.message,
-                            canPurchase = response.result.canLicense;
+                            canPurchase = response.result.canLicense,
+                            buyCreditsUrl = this.preview().buyCreditsUrl,
+                            displayFieldName = !this.isDownloaded() ? '<b>' + $.mage.__('File Name') + '</b>' : '';
 
                         this.getPrompt(
                             {
-                                'title': $.mage.__('License Adobe Stock Image?'),
-                                'content': '<p>' + confirmationContent + '</p><p><b>' + quotaMessage + '</p><br>' +
-                                           $.mage.__('File Name') + '</b>',
-                                'visible': canPurchase,
+                                'title': $.mage.__('License Adobe Stock Images?'),
+                                'content': '<p>' + confirmationContent + '</p><p><b>' + quotaMessage + '</b></p><br>' + displayFieldName,
+                                'visible': !this.isDownloaded(),
                                 'actions': {
                                     confirm: function (fileName) {
```

Comment:
`fileName` is `undefined` if input is not shown. The default generated filename should be used in that case, see `this.generateImageName(this.preview().displayedRecord())`

Reply:
I have fixed the issue but i'm not really sure about my approach. Can you please check the code. Thank you.